core/src/main/java/org/apache/brooklyn/core/mgmt/rebind/RebindIteration.java

```diff
@@ -687,8 +689,13 @@ protected void manageTheObjects() {
         Set<String> oldEntities = Sets.newLinkedHashSet(entityManager.getEntityIds());
         for (Entity entity: rebindContext.getEntities()) {
             ManagementTransitionMode oldMode = updateTransitionMode(entityManager, entity);
-            if (oldMode!=null)
+            if (oldMode!=null) {
                 oldEntities.remove(entity.getId());
+            }
+
+            if (ServiceStateLogic.getExpectedState(entity) == Lifecycle.STARTING || ServiceStateLogic.getExpectedState(entity) == Lifecycle.STOPPING) {
```

Comment:
See https://github.com/iyovcheva/brooklyn-server/pull/5 for an alternative (and for additional tests, including for hotStandby).